rdkit/Chem/PandasTools.py

```diff
@@ -549,108 +589,113 @@ def SaveXlsxFromFrame(frame, outFile, molCol='ROMol', size=(300, 300)):
+    worksheet.insert_image(c, 0, "f", {'image_data': image_data})
 
-    # Write first row with column names
     c2 = 1
     for x in cols:
-        worksheet.write_string(0, c2, x)
-        c2 += 1
-
-    c = 1
-    for _, row in frame.iterrows():
-        image_data = BytesIO()
-        img = Draw.MolToImage(row[molCol], size=size)
-        img.save(image_data, format='PNG')
-
-        worksheet.set_row(c, height=size[1])  # looks like height is not in px?
-        worksheet.insert_image(c, 0, "f", {'image_data': image_data})
-
-        c2 = 1
-        for x in cols:
-            if str(dataTypes[x]) == "object":
-                # string length is limited in xlsx
-                worksheet.write_string(c, c2, str(row[x])[:32000])
-            elif ('float' in str(dataTypes[x])) or ('int' in str(dataTypes[x])):
-                if (row[x] != np.nan) or (row[x] != np.inf):
-                    worksheet.write_number(c, c2, row[x])
-            elif 'datetime' in str(dataTypes[x]):
-                worksheet.write_datetime(c, c2, row[x])
-            c2 += 1
-        c += 1
-
-    workbook.close()
-    image_data.close()
+      if str(dataTypes[x]) == "object":
+        # string length is limited in xlsx
+        worksheet.write_string(c, c2, str(row[x])[:32000])
+      elif ('float' in str(dataTypes[x])) or ('int' in str(dataTypes[x])):
+        if (row[x] != np.nan) or (row[x] != np.inf):
+          worksheet.write_number(c, c2, row[x])
+      elif 'datetime' in str(dataTypes[x]):
+        worksheet.write_datetime(c, c2, row[x])
+      c2 += 1
+    c += 1
+
+  workbook.close()
+  image_data.close()
 
 
 def FrameToGridImage(frame, column='ROMol', legendsCol=None, **kwargs):
-    '''
+  '''
     Draw grid image of mols in pandas DataFrame.
     '''
-    if legendsCol:
-        if legendsCol == frame.index.name:
-            kwargs['legends'] = [str(c) for c in frame.index]
-        else:
-            kwargs['legends'] = [str(c) for c in frame[legendsCol]]
-    return Draw.MolsToGridImage(list(frame[column]), **kwargs)
+  if legendsCol:
+    if legendsCol == frame.index.name:
+      kwargs['legends'] = [str(c) for c in frame.index]
+    else:
+      kwargs['legends'] = [str(c) for c in frame[legendsCol]]
+  return Draw.MolsToGridImage(list(frame[column]), **kwargs)
 
 
 def AddMurckoToFrame(frame, molCol='ROMol', MurckoCol='Murcko_SMILES', Generic=False):
-    '''
+  '''
     Adds column with SMILES of Murcko scaffolds to pandas DataFrame.
 
     Generic set to true results in SMILES of generic framework.
     '''
-    if Generic:
-        def func(x): return Chem.MolToSmiles(MurckoScaffold.MakeScaffoldGeneric(
-          MurckoScaffold.GetScaffoldForMol(x[molCol])))
-    else:
-        def func(x): return Chem.MolToSmiles(MurckoScaffold.GetScaffoldForMol(x[molCol]))
-    frame[MurckoCol] = frame.apply(func, axis=1)
+  if Generic:
+
+    def func(x):
+      return Chem.MolToSmiles(
```

Comment:
future note:  might be nice to enable usage of the newer hashing code.

Reply:
Another good one for an enhancement PR